perpVects/QFPHelpers.cpp

```diff
@@ -37,19 +37,51 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
   return os;
 }
 
-std::string
-getSortName(sort_t val){
-  switch(val){
-  case lt:
-    return "lt";
-  case gt:
-    return "gt";
-  case bi:
-    return "bi";
-  case def:
-    return "us";
-  default:
-    return "something bad happened, undefined sort type";
+namespace FPHelpers{
+    auto
+  swap_float_int(unsigned int val) -> float{
```

Comment:
For the auto:
I tried the change, and no warning.  It's common to have different return types as long as the arguments are different types.

In this context, auto does NOT actually mean type deduction.  Instead it allows you to specify a "trailing return type".  The only reason that is ever useful is in using decltype() with some expression dependent on the arguments passed in, which is only useful for templates.  There's a section on it starting on page 25 of my "Effective Modern C++" book if you'd like to take a look next time you're in.